### ISSUE TITLE
[Feature Request/Bug] Support gradle 3.0.0-x for Android

### ISSUE THREAD

**Issue by AllanWang** [jitpack/jitpack.io#1634]
I've updated one of my projects to `gradle-3.0.0-alpha9`, and it looks like the submodules aren't working when they have api dependencies.

When using one of my submodules, everything builds and I can access the code, but I can't access any of the dependencies of that submodule. This wasn't an issue before hand with gradle 2.x.x and `compile` rather than `api`.

Looking at the [documentation](https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html), `api` should serve the same purchase as the now deprecated `compile`, and given that the modules work in the sample application (which implements the projects directly rather than through jitpack), this might be an issue with how jitpack builds the modules.

References to the project in question:

* [Jitpack builds](https://jitpack.io/#ca.allanwang/kau) to see older builds
* Three part build gradle for one of the submodules: [header](https://github.com/AllanWang/KAU/blob/canary/android-lib.gradle), [core](https://github.com/AllanWang/KAU/blob/canary/core/build.gradle), [artifacts](https://github.com/AllanWang/KAU/blob/canary/artifacts.gradle) 
* [Library test branch](https://github.com/AllanWang/KAU/tree/canary)
* [My stack overflow question](https://stackoverflow.com/questions/45447184/library-dependency-issues-with-gradle-3-0-0-alphax)
* [Build gradle](https://github.com/AllanWang/Frost-for-Facebook/blob/canary/app/build.gradle) of a separate app that takes the dependencies through jitpack.

Edit: Looks like there's an issue with loading dependencies

The [log](https://jitpack.io/ca/allanwang/kau/157d78c/build.log) builds but there are a lot of issues

**Comment by jitpack-io** [comment]
Hi,

Looks like this is still an open issue in the android-maven-gradle-plugin: https://github.com/dcendents/android-maven-gradle-plugin/issues/61

**Comment by AllanWang** [comment]
@jitpack-io thanks for the reference. I will be keeping watch on that issue. However, looking back at the logs, I find it weird that the gradle 3.0.0-x errors are actually present in the grade 2.3.3 versions, though those builds execute properly.

All of them have the following:

```java
/home/jitpack/build/adapter/src/main/kotlin/ca/allanwang/kau/animators/BaseItemAnimator.java:29: error: package android.support.annotation does not exist
import android.support.annotation.NonNull;
                                 ^
/home/jitpack/build/adapter/src/main/kotlin/ca/allanwang/kau/animators/BaseItemAnimator.java:30: error: package android.support.v4.view does not exist
import android.support.v4.view.ViewCompat;
                              ^
/home/jitpack/build/adapter/src/main/kotlin/ca/allanwang/kau/animators/BaseItemAnimator.java:31: error: package android.support.v7.widget.RecyclerView does not exist
import android.support.v7.widget.RecyclerView.ViewHolder;
                                             ^
/home/jitpack/build/adapter/src/main/kotlin/ca/allanwang/kau/animators/BaseItemAnimator.java:32: error: package android.support.v7.widget does not exist
import android.support.v7.widget.SimpleItemAnimator;
```

Unsure if that is also in relation to the maven plugin.

For reference:

* log [v3.2.1](https://jitpack.io/ca/allanwang/kau/3.2.1/build.log), which is a working build - [github link](https://github.com/AllanWang/KAU/tree/3.2.1)
* log [157d78c](https://jitpack.io/ca/allanwang/kau/157d78c/build.log), which was built with gradle 3.0.0-alpha9 - [github link](https://github.com/AllanWang/KAU/tree/157d78cae43c88cea11ac64a9df7e0d36f067a68)
* log [598ef8d](https://jitpack.io/ca/allanwang/kau/598ef8d/build.log), which was built with gradle 2.3.3 after I tried 3.0.0-alpha9 - [github link](https://github.com/AllanWang/KAU/tree/598ef8d98ee29bd9c1f288a980849d04d0b49583)

**Comment by jitpack-io** [comment]
Looks like the android-maven-gradle-plugin now detects api/implementation dependencies. This is still work in progress but available on 'develop' branch:
https://jitpack.io/#dcendents/android-maven-gradle-plugin/d4c1a63

**Comment by AllanWang** [comment]
Thanks for the heads up!

**Comment by AllanWang** [comment]
KAU has [built successfully](https://jitpack.io/ca/allanwang/kau/705423a/build.log)

In the process, there was a null exception with the `WindowsPathParser`, which seems to be fixed like [so](https://stackoverflow.com/questions/45597661/gradle-error-upgrading-to-android-studio-3-0-beta-1)

**Comment by AllanWang** [comment]
Even though it compiles, it looks like the `api` calls aren't acknowledged as api's. All dependencies listed as such won't be visible to apps that use the lib, so this is not yet a fully working solution.

I will continue monitoring the gradle dependency.

**Comment by AllanWang** [comment]
@jitpack-io dcendants has released v2.0 of the plugin
